minio/minio
# Add support for existing object replication.

## Description
Allow replication of older versioned content in a cluster (created prior to turning on replication/enabling replication rules) - this implements the S3 spec for existing object replication between 2 MinIO clusters with a replication setup.

Add support for a MinIO Extension API ResetBucketReplication - which will trigger a resync of previously replicated content from source to target. This is designed to cater to the use case where DR site is entirely lost and previously replicated content needs to be rehydrated on a new cluster

## Motivation and Context
https://aws.amazon.com/blogs/storage/replicating-existing-objects-between-s3-buckets/

## How to test this PR?
Dependencies: https://github.com/minio/madmin-go/pull/6
1. start 2 minio instances and copy content to a bucket on minio1
2. Set up replication target from minio1 -> minio2 with `mc admin bucket remote add`
3. Set up replication config with existing object replication enabled using `mc replicate add minio1/bucket --replicate "existing-objects..." --arn `
4. objects created prior to enabling replication or while replication rule is disabled should still sync when scanner runs

replication reset
1. `mc rm --versions --force --r minio2/bucket` to remove all content on target
2. `mc replicate reset minio1/bucket` should re-replicate the content to minio2 with same replication setup as above i.e. with existing object replication enabled.
## Types of changes

## Checklist:
